### ISSUE TITLE
Implement AMX support 

### ISSUE THREAD

**Issue by konstantin-s-bogom** [google/gvisor#9896]
### Description

Enabling AMX within gVisor needs at least the following: 
- Handling user calls to arch_prctl to enable it. 
- Adjusting signal stack sizes to support the extra 8256 bytes necessary for AMX. 

Currently it is fully disabled on host kernels that enable it.

**Comment by bbhattar** [comment]
@konstantin-s-bogom 
Is this still needed and do you still need help for enabling AMX? Intel can jump in if yes

**Comment by konstantin-s-bogom** [comment]
Yes this is still necessary; currently we unconditionally keep it disabled it everywhere. 

Implementing it will be most awkward for Systrap, where we currently have static mappings for sharing context state for user processes.
